File: 0108-convert-sorted-array-to-binary-search-tree/0108-convert-sorted-array-to-binary-search-tree-Solution1.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public TreeNode SortedArrayToBST(int[] nums) {
        return SortedArrayToBST2(nums, 0, nums.Length - 1);
    }

    public TreeNode SortedArrayToBST2(int[] nums, int left, int right) {
        if(left > right) return null;
        // if(left == right) return new TreeNode(nums[left]);
        var middle = left + (right - left) / 2;
        return new TreeNode(nums[middle], 
            SortedArrayToBST2(nums, left, middle - 1), 
            SortedArrayToBST2(nums, middle+1, right));
    }
}

File: 0108-convert-sorted-array-to-binary-search-tree/0108-convert-sorted-array-to-binary-search-tree-Solution1.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {number[]} nums
 * @return {TreeNode}
 */
var sortedArrayToBST = function(nums) {
    return sortedArrayToBST2(nums, 0, nums.length-1);
};

const sortedArrayToBST2 = (nums, left, right) => {
    if(left > right) return null;
    const middle = Math.floor(left + (right - left) / 2);
    return new TreeNode(nums[middle], sortedArrayToBST2(nums, left, middle-1), sortedArrayToBST2(nums, middle+1, right));
};